### ISSUE TITLE
stack setup fails during sanity check on Windows 10 2004

### ISSUE THREAD

**Issue by nchengyeeshen** [commercialhaskell/stack#5313]
## Description

`stack setup` successfully installs GHC but fails during the linking step of the sanity check. I can run `ghci` but am unable to install other packages such as `ghcid`

### Steps to reproduce

1. Download stack installer for Windows from https://get.haskellstack.org/stable/windows-x86_64-installer.exe
2. Install stack with installer
3. Open powershell and run `stack setup`

### Expected

Stack setup is successful.

### Actual

Stack setup successfully installs but fails during the sanity check. It fails on the linking step.

```
PS C:\Users\Nicholas> stack setup --reinstall
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
Already downloaded.
Already downloaded.
Already downloaded.
Decompressing ghc-8.8.3.tar.xz...

7-Zip 9.20  Copyright (c) 1999-2010 Igor Pavlov  2010-11-18

Processing archive: C:\Users\Nicholas\AppData\Local\Programs\stack\x86_64-windows\ghc-8.8.3.tar.xz

Extracting  ghc-8.8.3.tar

Everything is Ok

Size:       2555330560
Compressed: 204702116
Extracting ghc-8.8.3.tar...
Extracted total of 10091 files from ghc-8.8.3.tar
GHC installed to C:\Users\Nicholas\AppData\Local\Programs\stack\x86_64-windows\ghc-8.8.3\
The GHC located at C:\Users\Nicholas\AppData\Local\Programs\stack\x86_64-windows\ghc-8.8.3\bin\ghc-8.8.3.EXE failed to compile a sanity check. Please see:

    http://docs.haskellstack.org/en/stable/install_and_upgrade/

for more information. Exception was:
Received ExitFailure 11 when running
Raw command: "C:\\Users\\Nicholas\\AppData\\Local\\Programs\\stack\\x86_64-windows\\ghc-8.8.3\\bin\\ghc-8.8.3.exe" "C:\\Users\\Nicholas\\AppData\\Local\\Temp\\stack-sanity-check-fb68206da0544499\\Main.hs" -no-user-package-db
Run from: C:\Users\Nicholas\AppData\Local\Temp\stack-sanity-check-fb68206da0544499\
Standard output:

[1 of 1] Compiling Main             ( C:\\Users\Nicholas\AppData\Local\Temp\stack-sanity-check-fb68206da0544499\Main.hs, C:\\Users\Nicholas\AppData\Local\Temp\stack-sanity-check-fb68206da0544499\Main.o )
Linking C:\\Users\Nicholas\AppData\Local\Temp\stack-sanity-check-fb68206da0544499\Main.exe ...
Standard error:


Access violation in generated code when writing 0x0

 Attempting to reconstruct a stack trace...

   Frame        Code address
 * 0x461da80    0x37f7b66 C:\Users\Nicholas\AppData\Local\Programs\stack\x86_64-windows\ghc-8.8.3\bin\ghc.exe+0x33f7b66
 * 0x461da88    0x3277bb9 C:\Users\Nicholas\AppData\Local\Programs\stack\x86_64-windows\ghc-8.8.3\bin\ghc.exe+0x2e77bb9
 * 0x461da90    0x5
 * 0x461da98    0x5
 * 0x461daa0    0x3d7dc22 C:\Users\Nicholas\AppData\Local\Programs\stack\x86_64-windows\ghc-8.8.3\bin\ghc.exe+0x397dc22
 * 0x461daa8    0x762d270
```

### Stack version

```
PS C:\Users\Nicholas> stack --version
Version 2.3.1, Git revision de2a7b694f07de7e6cf17f8c92338c16286b2878 (8103 commits) x86_64 hpack-0.33.0
```

### Method of installation

* Official binary, downloaded from stackage.org or fpcomplete's package repository

### Other notes

I've also tried `stack setup` with `%STACKROOT%` set to `C:\sr`  and running it from an elevated prompt. It's still failing at the same stage.

**Comment by gdziadkiewicz** [comment]
This looks like https://gitlab.haskell.org/ghc/ghc/issues/17926#note_266472 and seems to confirm what the comment states. Can someone more familiar with the way changes flow around GHC and end in Stack tell when can we expect this fix to be available via Stack? I'm personally affected and willing to help, if possible.

**Comment by NikolaiPohodenko** [comment]
another way to reproduce:

1. download stack installer and install stack with default install options
2. create a new project by stack new  proj-15.4 --resolver=lts-15.4
3. try to build it. The build will fail identically.
4. build will also fail on all LTS that use GHC 8.8.3, luckily those LTSs that use GHC 8.8.2 are ok.

I suggest, that on Stack part, at least a minimal automatic smoke test is required.

**Comment by nchengyeeshen** [comment]
I'm closing this issue since stack now resolves to a GHC 8.8.4 which doesn't have this problem. See [this](https://github.com/commercialhaskell/stack/issues/5347#issuecomment-696346250) as well.
